biodatasets/pubhealth/pubhealth.py

```diff
@@ -0,0 +1,183 @@
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls = _URLs[_DATASETNAME]
+        data_dir = Path(dl_manager.download_and_extract(urls))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/train.tsv"), "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/test.tsv"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"filepath": os.path.join(data_dir, "PUBHEALTH/dev.tsv"), "split": "validation"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """Yields examples as (key, example) tuples."""
+
+        with open(filepath, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(
+                csv_file, quotechar='"', delimiter="\t", quoting=csv.QUOTE_ALL, skipinitialspace=True
+            )
+            next(csv_reader, None)  # remove column headers
+            for id_, row in enumerate(csv_reader):
+                # train.tsv/dev.tsv only has 9 columns
+                # test.tsv has an additional column at the beginning
+                #  Some entries are malformed, will log skipped lines
+                if len(row) < 9:
+                    logger.warning("Line %s is malformed", id_)
+                    continue
+                (
+                    claim_id,
+                    claim,
+                    date_published,
+                    explanation,
+                    fact_checkers,
+                    main_text,
+                    sources,
+                    label,
+                    subjects,
+                ) = row[
+                    -9:
+                ]  # only take last 9 columns to fix test.tsv disparity
+
+                if self.config.schema == "source":
+                    yield id_, {
+                        "claim_id": claim_id,
+                        "claim": claim,
+                        "date_published": date_published,
+                        "explanation": explanation,
+                        "fact_checkers": fact_checkers,
+                        "main_text": main_text,
+                        "sources": sources,
+                        "label": label,
+                        "subjects": subjects,
+                    }
+
+                elif self.config.schema == "bigbio_text":
+                    yield id_, {
+                        "id": id_,  # uid is an unique identifier for every record that starts from 0
+                        "document_id": claim_id,
+                        "text": main_text,
+                        "labels": [label],
```

Comment:
I'm just browsing the paper associated with this task and I'm not sure it makes sense as a (main_text, label) task. It seems that the claim, explanation, and main_text need to work together somehow to produce the label. What do you think?

Reply:
Good point. I think if you look at figure 1 in the paper the text classification phase (veracity) utilizes both the claim and the explanation. So maybe we can create the "text" field to capture both by inserting something like: 
f"CLAIM: {claim} EXPLANATION: {explanation}"

Reply:
Does that sound good or do you have other ideas?

Reply:
Yes, I can see it. I'll adjust the bigbio_schema and resubmit.

Reply:
did you put some thought into this part?
```
if we cant figure out a mapping between the existing labels [true, false, unproven, mix] and [entailment, neutral, contradiction]
```

Reply:
Apologies, I totally misunderstood what you meant in that. I don't think we can make this an entailment task and instead need to make it a pairs task, since the explanation is used to verify the claim. In the instances the explanation is used to explain the label for the hypothesis. So I don't think that actually fits entailment.

Reply:
No problem, I think you could make a case for mapping `true` to `entailment`, `false` to `contradiction` and collapsing `unproven` and `mix` into `neutral` but I think the text pairs version retains more of the info in the dataset so I think i'm happy either way. If you want to implement as text pairs with 4 possible values for the label that works for me. ps consider using  https://huggingface.co/docs/datasets/v2.0.0/en/package_reference/main_classes#datasets.ClassLabel in the source schema to indicate the 4 possible values (an example from BigBio https://github.com/bigscience-workshop/biomedical/blob/beac33de5b4044d718e3719832053a4315c24da8/biodatasets/meddialog/meddialog.py#L136)